### PR TITLE
Partition load_state_dict child keys once

### PR DESCRIPTION
**Pull request by Rxflex**
## Summary
- partition child state_dict entries in a single pass per module during load_state_dict
- avoid scanning all local state_dict keys once per child module
- add a regression test that detects cross-sibling prefix checks

## Test plan
- python -m py_compile torch\\nn\\modules\\module.py test\\nn\\test_load_state_dict.py
- git diff --check

Not run locally: full test/nn/test_load_state_dict.py requires an editable PyTorch develop build with torch._C available in this checkout. lintrunner is not installed in the current Python environment.

### CODE REVIEW

**Comment by Rxflex** [comment]
﻿@pytorchbot label "topic: not user facing"

**Comment by Rxflex** [comment]
﻿Added a standalone benchmark for the optimized path:

```
python benchmarks/serialization/load_state_dict_bench.py --children 128 512 2048
python benchmarks/serialization/load_state_dict_bench.py --children 128 512 2048 --assign
```

The benchmark keeps tensors small so it measures Python key routing in `load_state_dict` rather than tensor copy bandwidth.

Additional local checks:

```
python -m py_compile benchmarks\serialization\load_state_dict_bench.py
python D:\code\pytorch\benchmarks\serialization\load_state_dict_bench.py --children 4 8 --min-run-time 0.01
```

The sanity benchmark command was run from outside the checkout and therefore used the installed PyTorch wheel, only to validate the script execution path.

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -2594,14 +2594,22 @@ def load(module, local_state_dict, prefix="") -> None:
...
             )
+            child_state_dicts = {
+                name: {}
+                for name, child in module._modules.items()
+                if child is not None
+            }
+            for key, value in local_state_dict.items():
+                if not key.startswith(prefix):
+                    continue
+                input_name = key[len(prefix) :].split(".", 1)
```
Comment (Skylion007) [inline torch/nn/modules/module.py]:
```suggestion
                input_name = key.removeprefix(prefix).split(".", 1)

Comment (Rxflex) [reply torch/nn/modules/module.py]: hi, thx, fixed :)
